--- origsrc/gtk-vnc-0.3.10/src/Makefile.am	2009-09-17 08:16:10.000000000 -0500
+++ src/gtk-vnc-0.3.10/src/Makefile.am	2009-10-28 14:21:19.858180100 -0500
@@ -80,10 +80,10 @@ CLEANFILES = vncmarshal.c vncmarshal.h e
 if WITH_PYTHON
 pyexec_LTLIBRARIES = gtkvnc.la
 
-gtkvnc_la_LIBADD = libgtk-vnc-1.0.la @PYGTK_LIBS@
+gtkvnc_la_LIBADD = libgtk-vnc-1.0.la @PYGTK_LIBS@ -lpython$(PYTHON_VERSION)
 gtkvnc_la_CFLAGS = @GTK_CFLAGS@ @PYTHON_INCLUDES@ \
 		   @PYGTK_CFLAGS@ @WARNING_PYTHON_CFLAGS@
-gtkvnc_la_LDFLAGS = -module -avoid-version -fPIC
+gtkvnc_la_LDFLAGS = -module -avoid-version -no-undefined
 gtkvnc_la_SOURCES = vncmodule.c vncmodule.defs.c
 
 CODEGENDIR = $(shell pkg-config --variable=codegendir pygtk-2.0)
