--- origsrc/gtk-vnc-0.4.4/configure.ac	2011-11-10 09:38:35.000000000 -0600
+++ src/gtk-vnc-0.4.4/configure.ac	2011-11-16 13:53:07.936955100 -0600
@@ -146,7 +146,7 @@ case "$host" in
     ;;
 
   *-*-cygwin*)
-    NO_UNDEFINED_LDFLAGS="-no-undefined"
+    NO_UNDEFINED_FLAGS="-no-undefined"
     ;;
 
   *)
--- origsrc/gtk-vnc-0.4.4/src/Makefile.am
+++ src/gtk-vnc-0.4.4/src/Makefile.am
@@ -255,10 +255,10 @@ CLEANFILES = vncmarshal.c vncmarshal.h \
 if WITH_PYTHON
 pyexec_LTLIBRARIES = gtkvnc.la

-gtkvnc_la_LIBADD = libgtk-vnc-$(GTK_VNC_API_VERSION).la @PYGTK_LIBS@
+gtkvnc_la_LIBADD = libgtk-vnc-$(GTK_VNC_API_VERSION).la @PYGTK_LIBS@ -lpython$(PYTHON_VERSION)
 gtkvnc_la_CFLAGS = @GTK_CFLAGS@ @PYTHON_INCLUDES@ \
 		   @PYGTK_CFLAGS@ @WARNING_PYTHON_CFLAGS@
-gtkvnc_la_LDFLAGS = -module -avoid-version -fPIC
+gtkvnc_la_LDFLAGS = -module -avoid-version $(NO_UNDEFINED_FLAGS)
 gtkvnc_la_SOURCES = vncmodule.c
 nodist_gtkvnc_la_SOURCES = vncmodule.defs.c
 BUILT_SOURCES += vncmodule.defs.c
